mitmproxy/tools/web/app.py

```diff
@@ -34,6 +34,12 @@ def flow_to_json(flow: mitmproxy.flow.Flow) -> dict:
         "type": flow.type,
         "modified": flow.modified(),
     }
+    # .alpn_proto_negotiated is bytes, we need to decode that.
+    for conn in "client_conn", "server_conn":
+        if f[conn]["alpn_proto_negotiated"] is None:
+            continue
+        f[conn]["alpn_proto_negotiated"] = \
+            f[conn]["alpn_proto_negotiated"].decode(errors="backslashreplace")
```

Comment:
This is only what is sent to the web UI (which only reads it, never writes it), the actual flow remains untouched. I think that's okay to not be overprecise here? A non-ascii protocol would be suspicious by itself.